api/routers/metrics/area_metrics.py

```diff
@@ -45,8 +45,8 @@ def get_area_occupancy_daily_report(areas: str = "",
 def get_area_occupancy_weekly_report(
         areas: str = "",
         weeks: int = Query(0),
-        from_date: date = Query((date.today() - timedelta(days=date.today().weekday(), weeks=4)).isoformat()),
```

Comment:
Why are we removing this?

Reply:
Because these causes several bugs.

First, when you do not send neither from_date nor the to_date as a query parameter, default values will take place and will be sent to the corresponding report function. However, in both cases, a string will be sent, not a date. (Important: `.isoformat() `returns an string).
So, for example, when `get_weekly_metric` is reached and we have to set a week_span:
```
        if number_of_days > 0:
            # Separate weeks in range taking a number of weeks ago, considering the week ended yesterday
            date_range = pd.date_range(end=date.today() - timedelta(days=1), periods=number_of_days)
            start_dates = date_range[0::7]
            end_dates = date_range[6::7]
            week_span = list(zip(start_dates, end_dates))
        elif isinstance(from_date, date) and isinstance(to_date, date):
            # Separate weeks in range considering the week starts on Monday
            date_range = pd.date_range(start=from_date, end=to_date)
            week_span = list(parse_date_range(date_range))
        else:
            week_span = []
```
Given range of dates are ok, but are both strings, so will never enter `elif isinstance(from_date, date) and isinstance(to_date, date):` and week_span will be an empty array, and this, should not be like that. As a result, no data will be returned on the endpoint, regardless of whether or not there is.

Secondly, there was another issue that appears when you submit only one the two dates a let the other one be the default value. As we explained before, default values will be string, while submitted dates are type: "date". So when the code reaches:
```
def validate_dates(from_date: date, to_date: date):
    if from_date > to_date:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=bad_request_serializer(
                "Invalid range of dates",
                error_type="from_date doesn't come before to_date",
                loc=["query", "from_date"]
            )
        )
```
An exception will be thrown because we cannot compare a string with a date.